.nvidia-ci.yml

```diff
@@ -244,3 +244,60 @@ release:ngc-packaging:
   extends:
     - .dist-packaging
     - .release:ngc
+
+# Define the external image signing steps for NGC
+# Download the ngc cli binary for use in the sign steps
+.ngccli-setup:
+  before_script:
+    - export NGCLI_VERSION=3.41.4
+    - apt-get update && apt-get install -y curl unzip
+    - curl -sSLo ngccli_linux.zip https://api.ngc.nvidia.com/v2/resources/nvidia/ngc-apps/ngc_cli/versions/${NGCLI_VERSION}/files/ngccli_linux.zip
+    - unzip ngccli_linux.zip
+    - chmod u+x ngc-cli/ngc
+    - export PATH=$(pwd)/ngc-cli:${PATH}
+    - NGC_CLI_API_KEY=${NGC_REGISTRY_TOKEN} ngc version upgrade --org nvidia -y
+
+# .sign forms the base of the deployment jobs which signs images in the CI registry.
+# This is extended with the image name and version to be deployed.
+.sign:ngc:
+  image: ubuntu:latest
+  stage: sign
+  rules:
+    - if: $CI_COMMIT_TAG
+  variables:
+    IMAGE_NAME: "${NGC_REGISTRY_IMAGE}"
+    IMAGE_TAG: "${CI_COMMIT_TAG}"
+  retry:
+    max: 2
+  before_script:
+    - !reference [.ngccli-setup, before_script]
+    - |
+      if [ -n "$DIST" ]; then
+        export IMAGE_TAG="${IMAGE_TAG}-${DIST}"
+      fi
```

Comment:
When is this condition ever true? We seem to always use this where `DIST` is set.

Let's rather remove this and specify `IMAGE_TAG` as `${CI_COMMIT_TAG}-${DIST}` for other cases where we have the "short" tag, we can override ONLY `IMAGE_TAG` in the job that extends this one.